mycore-ocfl/src/main/java/org/mycore/ocfl/MCROCFLPersistenceTransaction.java

```diff
@@ -0,0 +1,165 @@
+
+    private final boolean verboseLogging = MCRConfiguration2.getBoolean("MCR.OCFL.PT.Verbose").orElse(false);
+
+    public MCROCFLPersistenceTransaction() {
+        try {
+            managerOpt = MCRConfiguration2
+                .<MCROCFLXMLClassificationManager>getSingleInstanceOf("MCR.Classification.Manager");
+        } catch (Exception e) {
+            if (verboseLogging) {
+                LOGGER.debug("ClassificationManager could not be found, setting to empty.");
+            }
+            managerOpt = Optional.empty();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isReady() {
+        if (verboseLogging) {
+            LOGGER.debug("TRANSACTION READY CHECK - {}", managerOpt.isPresent());
+        }
+        if (!managerOpt.isPresent()) {
+            return false;
+        } else {
+            return !isActive() && managerOpt.get().isMutable();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void begin() {
+        if (verboseLogging) {
+            LOGGER.debug("TRANSACTION BEGIN");
+        }
+        // if (isActive()) {
+        //     throw new IllegalStateException("TRANSACTION ALREADY ACTIVE");
+        // }
+        if (isActive()) {
+            LOGGER.warn("EXISTING TRANSACTION, ROLLING BACK FOR CLEAN STATE");
+            rollback();
+        }
+        currentSession = MCRSessionMgr.getCurrentSession();
+        currentSession.put("classQueue", new ArrayList<MCREvent>());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public void commit() {
+        if (verboseLogging) {
+            LOGGER.debug("TRANSACTION COMMIT");
+        }
+        if (!isActive() || getRollbackOnly()) {
+            throw new IllegalStateException("TRANSACTION NOT ACTIVE OR ONLY ROLLBACK");
+        }
+        try {
+            managerOpt.get().commitSession(currentSession);
+        } catch (Exception e) {
+            rollbackOnly = true;
+            rollbackList = (ArrayList<MCREvent>) currentSession.get("classQueue");
+            throw e;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public void rollback() {
+        if (verboseLogging) {
+            LOGGER.debug("TRANSACTION ROLLBACK");
+        }
+        if (!isActive()) {
+            throw new IllegalStateException("TRANSACTION NOT ACTIVE");
+        }
+        try {
+            if (rollbackOnly) {
+                ((ArrayList<MCREvent>) currentSession.get("classQueue")).addAll(rollbackList);
+            }
+        } catch (Exception e) {
+            currentSession.put("classQueue", rollbackList);
+        }
+        managerOpt.get().rollbackSession(currentSession);
+        rollbackOnly = false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean getRollbackOnly() {
+        if (verboseLogging) {
```

Comment:
LOGGER.debug already checks if Specific logger is on debug level. No need to specify any property or adding custom checks